File: src/lib/localization.ts
import { Place, RecommendationPlace } from './api'

export function getLocalizedName(place: Place | RecommendationPlace, locale: string): string {
  if (locale === 'ar' && place.name_ar) return place.name_ar
  if (locale === 'en' && place.name_en) return place.name_en
  return place.name
}

export function getLocalizedDescription(place: Place, locale: string): string {
  if (locale === 'ar' && place.description_ar) return place.description_ar
  if (locale === 'en' && place.description_en) return place.description_en
  return place.description
}

File: backend/app/utils/localization.py
from typing import Any

def get_preferred_language(accept_language: str | None) -> str:
    """Determine preferred language from Accept-Language header."""
    if not accept_language:
        return "en"
    
    # Simple parser for Accept-Language header
    # e.g., "ar-SA,ar;q=0.9,en-US;q=0.8,en;q=0.7"
    try:
        languages = [lang.split(";")[0].split("-")[0].strip().lower() for lang in accept_language.split(",")]
        for lang in languages:
            if lang in ["ar", "en"]:
                return lang
    except Exception:
        pass
    
    return "en"

def localize_place_schema(place_read_obj: Any, lang: str) -> Any:
    """Update name and description based on language preference."""
    if lang == "ar":
        if place_read_obj.name_ar:
            place_read_obj.name = place_read_obj.name_ar
        if place_read_obj.description_ar:
            place_read_obj.description = place_read_obj.description_ar
    elif lang == "en":
        if place_read_obj.name_en:
            place_read_obj.name = place_read_obj.name_en
        if place_read_obj.description_en:
            place_read_obj.description = place_read_obj.description_en
    return place_read_obj
